### PR TITLE
Remove `profile("training_step_and_backward")`

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?

Remove `profile("training_step_and_backward")` in `Closure` class. It is not needed since we already profile the `training_step` and `backward` calls individually. Also remove `_profiler` reference from the class

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by awaelchli — COMMENTED** [review]
> Remove TrainingTypePlugin.optimizer_zero_grad since no TTP subclasses this hook,

How do you know that? What would be the alternative? Wrap/patch the optimizer's zero_grad method? It's not clear to me why this removal is necessary

**Comment by daniellepintz** [comment]
> How do you know that?

I grep'ed all the TTP classes for `def optimizer_zero_grad` and none of them override it.

>  What would be the alternative? 

Users can override the LM's `optimizer_zero_grad` method

> It's not clear to me why this removal is necessary

This removal isn't necessary, it is a simplification.

File: pytorch_lightning/loops/optimization/optimizer_loop.py [hunk 1]
```diff
@@ -132,22 +132,18 @@ def __init__(
         self._step_fn = step_fn
```
Comment (carmocca) [inline pytorch_lightning/loops/optimization/optimizer_loop.py:132]: Remove the init argument and docstring (above)

File: pytorch_lightning/profiler/pytorch.py [hunk 2]
```diff
@@ -195,7 +195,6 @@ def __call__(self, num_step: int) -> "ProfilerAction":
 class PyTorchProfiler(BaseProfiler):
```
Comment (carmocca) [inline pytorch_lightning/profiler/pytorch.py:195]: There are still some mentions in the docs of
pytorch_lightning/profiler/__init__.py

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/profiler/pytorch.py (hunk 2 above)
Comment (daniellepintz) [reply pytorch_lightning/profiler/pytorch.py:195]: thanks, fixed!

File: pytorch_lightning/strategies/strategy.py [hunk 3]
```diff
@@ -180,10 +180,6 @@ def optimizer_step(
         model = model or self.lightning_module
         self.precision_plugin.optimizer_step(model, optimizer, opt_idx, closure, **kwargs)
 
-    def optimizer_zero_grad(self, current_epoch: int, batch_idx: int, optimizer: Optimizer, opt_idx: int) -> None:
-        """Zeros all model parameter's gradients."""
-        self.lightning_module.optimizer_zero_grad(current_epoch, batch_idx, optimizer, opt_idx)
-
```
Comment (ananthsub) [inline pytorch_lightning/strategies/strategy.py:186]: We could again split this out to another PR for discussion to ensure this PR does only one thing

Comment (daniellepintz) [reply pytorch_lightning/strategies/strategy.py]: Done

**Review by awaelchli — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -346,6 +346,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Removed `Strategy.init_optimizers` in favor of `Strategy.setup_optimizers` ([#11236](https://github.com/PyTorchLightning/pytorch-lightning/pull/11236))
 
+
+- Removed `profile("training_step_and_backward")` ([#11222](https://github.com/PyTorchLightning/pytorch-lightning/pull/11222))
```
Comment (Borda) [inline CHANGELOG.md:350]:
```suggestion
- Removed `profile("training_step_and_backward")` in `Closure` class as It is duplicate to profile calls `training_step` and `backward` ([#11222](https://github.com/PyTorchLightning/pytorch-lightning/pull/11222))
```

**Review by Borda — APPROVED** [review]

**Comment by daniellepintz** [comment]
Ready for merge!
